integrations/php/workload-executor.php

```diff
@@ -0,0 +1,111 @@
+<?php declare(strict_types=1);
+
+use MongoDB\Tests\UnifiedSpecTests\EntityMap;
+use MongoDB\Tests\UnifiedSpecTests\Loop;
+use MongoDB\Tests\UnifiedSpecTests\UnifiedTestCase;
+use MongoDB\Tests\UnifiedSpecTests\UnifiedTestRunner;
+use function MongoDB\BSON\fromJSON;
+use function MongoDB\BSON\toPHP;
+
+require_once __DIR__ . '/vendor/autoload.php';
+require_once __DIR__ . '/vendor/bin/.phpunit/phpunit/vendor/autoload.php';
+
+class Logger
+{
+    private $events = [];
+    private $errors = [];
+    private $failures = [];
+    private $numSuccesses = -1;
+    private $numIterations = -1;
+
+    public function applyEntityMap(EntityMap $entityMap) : void
+    {
+        if ($entityMap->offsetExists('events')) {
+            $this->events = count($this->events) > 0
+                ? array_merge($this->events, (array) $entityMap['events'])
+                : (array) $entityMap['events'];
```

Comment:
The count check isn't necessary, as `array_merge` with an empty array will produce the expected result. OTOH, this doesn't hurt, so I'll let you decide whether you want to make changes.

Reply:
Ah, I wasn't sure about the performance vs. a basic assignment, but it looks like [`php_array_merge_wrapper`](https://github.com/php/php-src/blob/php-8.0.3/ext/standard/array.c#L3818) will end up doing a `ZVAL_COPY` in the common case where `$this->events` is empty since `$entityMap['events']` should be packed.

Reply:
Ah, good to know this is actually founded in logic and not just ease to read. Thanks for checking the implementation!